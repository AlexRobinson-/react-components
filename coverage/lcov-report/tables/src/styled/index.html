
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for tables/src/styled</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> tables/src/styled</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.38% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>216/239</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.32% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>44/81</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>56/56</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>207/230</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="StyledBody.ts"><a href="StyledBody.ts.html">StyledBody.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledCaption.ts"><a href="StyledCaption.ts.html">StyledCaption.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledCell.ts"><a href="StyledCell.ts.html">StyledCell.ts</a></td>
	<td data-value="80" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="25" class="abs high">20/25</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="25" class="abs high">20/25</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledGroupRow.ts"><a href="StyledGroupRow.ts.html">StyledGroupRow.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledHead.ts"><a href="StyledHead.ts.html">StyledHead.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledHeaderCell.ts"><a href="StyledHeaderCell.ts.html">StyledHeaderCell.ts</a></td>
	<td data-value="91.67" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 91%"></div><div class="cover-empty" style="width: 9%"></div></div>
	</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="24" class="abs high">22/24</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="24" class="abs high">22/24</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledHeaderRow.ts"><a href="StyledHeaderRow.ts.html">StyledHeaderRow.ts</a></td>
	<td data-value="88.89" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="18" class="abs high">16/18</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="18" class="abs high">16/18</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledOverflowButton.ts"><a href="StyledOverflowButton.ts.html">StyledOverflowButton.ts</a></td>
	<td data-value="92.59" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledRow.ts"><a href="StyledRow.ts.html">StyledRow.ts</a></td>
	<td data-value="81.25" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 81%"></div><div class="cover-empty" style="width: 19%"></div></div>
	</td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="32" class="abs high">26/32</td>
	<td data-value="43.75" class="pct low">43.75%</td>
	<td data-value="16" class="abs low">7/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="32" class="abs high">26/32</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledSortableButton.ts"><a href="StyledSortableButton.ts.html">StyledSortableButton.ts</a></td>
	<td data-value="88.57" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.57" class="pct high">88.57%</td>
	<td data-value="35" class="abs high">31/35</td>
	<td data-value="59.09" class="pct medium">59.09%</td>
	<td data-value="22" class="abs medium">13/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="88.57" class="pct high">88.57%</td>
	<td data-value="35" class="abs high">31/35</td>
	</tr>

<tr>
	<td class="file high" data-value="StyledTable.ts"><a href="StyledTable.ts.html">StyledTable.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	</tr>

<tr>
	<td class="file medium" data-value="style-utils.ts"><a href="style-utils.ts.html">style-utils.ts</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Feb 19 2020 18:29:44 GMT+0000 (Coordinated Universal Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    