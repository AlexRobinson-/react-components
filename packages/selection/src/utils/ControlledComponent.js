import { PureComponent } from 'react';
import PropTypes from 'prop-types';

/** @component */
export default class ControlledComponent extends PureComponent {
  static propTypes = {
    onStateChange: PropTypes.func
  };

  isControlledProp = (key) => {
    return this.props.hasOwnProperty(key)
  }

  /**
   * Used to help retrieve state that can be controlled through props
   */
  getControlledState = (stateToMerge = this.state) => {
    return Object.keys(stateToMerge).reduce((state, key) => {
      state[key] = this.isControlledProp(key)
        ? this.props[key]
        : stateToMerge[key]
      return state
    }, {});
  };

  /**
   * Used to help set state that can be controlled through props
   */
  setControlledState = (newState = {}) => {
    const { onStateChange } = this.props;
    const currentState = this.getControlledState();

    if (onStateChange) {
      const newInternalState = {};
      Object.keys(newState).forEach(key => {
        if (currentState[key] !== newState[key]) {
          newInternalState[key] = newState[key];
        }
      });

      onStateChange(newInternalState);
    } else {
      this.setState(newState);
    }
  };
}
