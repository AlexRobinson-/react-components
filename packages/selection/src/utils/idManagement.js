let idCounter = 1;

/**
 * This generates a unique ID for all autocomplete inputs
 * @param {String} prefix the prefix for the id
 * @return {String} the unique ID
 *
 * @component
 */
function generateId(prefix = 'garden') {
  return `${prefix}-${idCounter++}`
}

/**
 * This is only used in tests... Could be useful in SSR?
 * @param {Number} num The number to set the idCountry to
 *
 * @component
 */
function setIdCounter(num) {
  idCounter = num
}

export default {
  generateId,
  setIdCounter
};
