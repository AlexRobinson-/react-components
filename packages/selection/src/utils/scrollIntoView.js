
function findParent(finder, node, rootNode) {
  if (node !== null && node !== rootNode.parentNode) {
    if (finder(node)) {
      if (node === document.body && node.scrollTop === 0) {
        // in chrome body.scrollTop always return 0
        return document.documentElement
      }
      return node
    } else {
      return findParent(finder, node.parentNode, rootNode)
    }
  } else {
    return null
  }
}

/**
 * Get the closest element that scrolls
 * @param {HTMLElement} node - the child element to start searching for scroll parent at
 * @param {HTMLElement} rootNode - the root element of the component
 * @return {HTMLElement} the closest parentNode that scrolls
 */
const getClosestScrollParent = findParent.bind(
  null,
  node => node.scrollHeight > node.clientHeight + 4,
)

/**
 * Scroll node into view if necessary
 * @param {HTMLElement} node - the element that should scroll into view
 * @param {HTMLElement} rootNode - the root element of the component
 */
// eslint-disable-next-line complexity
export default function scrollIntoView(node, rootNode) {
  const scrollParent = getClosestScrollParent(node, rootNode)
  if (scrollParent === null) {
    return
  }
  const scrollParentStyles = getComputedStyle(scrollParent)
  const scrollParentRect = scrollParent.getBoundingClientRect()
  const scrollParentBorderTopWidth = parseInt(
    scrollParentStyles.borderTopWidth,
    10,
  )
  const scrollParentBorderBottomWidth = parseInt(
    scrollParentStyles.borderBottomWidth,
    10,
  )
  const bordersWidth =
    scrollParentBorderTopWidth + scrollParentBorderBottomWidth
  const scrollParentTop = scrollParentRect.top + scrollParentBorderTopWidth
  const nodeRect = node.getBoundingClientRect()

  if (nodeRect.top < 0 && scrollParentRect.top < 0) {
    scrollParent.scrollTop += nodeRect.top
    return
  }

  if (nodeRect.top < 0) {
    // the item is above the viewport and the parent is not above the viewport
    scrollParent.scrollTop += nodeRect.top - scrollParentTop
    return
  }

  if (nodeRect.top > 0 && scrollParentRect.top < 0) {
    if (
      scrollParentRect.bottom > 0 &&
      nodeRect.bottom + bordersWidth > scrollParentRect.bottom
    ) {
      // the item is below scrollable area
      scrollParent.scrollTop +=
        nodeRect.bottom - scrollParentRect.bottom + bordersWidth
    }
    // item and parent top are on different sides of view top border (do nothing)
    return
  }

  const nodeOffsetTop = nodeRect.top + scrollParent.scrollTop
  const nodeTop = nodeOffsetTop - scrollParentTop
  if (nodeTop < scrollParent.scrollTop) {
    // the item is above the scrollable area
    scrollParent.scrollTop = nodeTop
  } else if (
    nodeTop + nodeRect.height + bordersWidth >
    scrollParent.scrollTop + scrollParentRect.height
  ) {
    // the item is below the scrollable area
    scrollParent.scrollTop =
      nodeTop + nodeRect.height - scrollParentRect.height + bordersWidth
  }
  // the item is within the scrollable area (do nothing)
}
