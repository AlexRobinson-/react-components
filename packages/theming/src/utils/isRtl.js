/** @component */
export default function isRtl(props) {
  return props.theme && props.theme.rtl;
}
