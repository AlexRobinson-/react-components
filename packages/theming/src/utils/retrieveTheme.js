/** @component */
export default function retrieveTheme(componentId, props) {
  const styles = props.theme.styles;

  if (!styles) {
    return;
  }

  const componentStyles = styles[componentId];
  if (typeof componentStyles === 'function') {
    return componentStyles(props);
  }

  return componentStyles;
}
