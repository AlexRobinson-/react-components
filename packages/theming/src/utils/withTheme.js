import { withTheme as styledWithTheme } from 'styled-components';

/** @component */
export default function withTheme(WrappedComonent) {
  return styledWithTheme(WrappedComonent);
};
