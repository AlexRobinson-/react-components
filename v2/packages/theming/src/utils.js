import { withTheme } from 'styled-components';

function retrieveTheme(componentId, props) {
    const styles = props.theme.styles;

    if (!styles) {
        return;
    }

    const componentStyles = styles[componentId];
    if (typeof componentStyles === 'function') {
        return componentStyles(props);
    }

    return componentStyles;
}

function isRtl(props) {
    return props.theme && props.theme.rtl;
}

export default {
    isRtl,
    withTheme,
    retrieveTheme
};
