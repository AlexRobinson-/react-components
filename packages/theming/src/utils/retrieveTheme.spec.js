import React from 'react';
import { mountWithTheme } from 'utils';
import retrieveTheme from './retrieveTheme';
import { shallow } from 'enzyme';
import styled from 'styled-components';

describe('retrieveTheme', () => {
  const Example = styled.div`
    ${props => retrieveTheme('example', props)};
  `;

  it('should not apply theme if no ThemeProvider is found', () => {
    const wrapper = shallow(<Example />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not apply theme if no matching ID is found', () => {
    const wrapper = mountWithTheme(<Example />, false, { 'test-id': 'color: red;' });

    expect(wrapper).toMatchSnapshot();
  });

  it('should apply theme if ID is found', () => {
    const wrapper = mountWithTheme(<Example />, false, { example: 'color: red;' });

    expect(wrapper).toMatchSnapshot();
  });
});
