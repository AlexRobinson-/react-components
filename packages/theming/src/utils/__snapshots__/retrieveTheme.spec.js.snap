// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`retrieveTheme should apply theme if ID is found 1`] = `
<retrieveThemespec__Example>
  <div
    className="retrieveThemespec__Example-ePzsat eRuwrd"
  />
</retrieveThemespec__Example>
`;

exports[`retrieveTheme should not apply theme if no ThemeProvider is found 1`] = `
<div
  className="retrieveThemespec__Example-ePzsat eRuwrd"
/>
`;

exports[`retrieveTheme should not apply theme if no matching ID is found 1`] = `
<retrieveThemespec__Example>
  <div
    className="retrieveThemespec__Example-ePzsat eRuwrd"
  />
</retrieveThemespec__Example>
`;
